### DISCUSSION TITLE
Alert user when the queryClient is configured in a way that it is shared between different users and requests

### DISCUSSION THREAD

**Discussion by josephfarina** [TanStack/query#3375] [Ideas]
Hello!

We recently just ran across a bug that was very difficult to track down. It was caused partially by initializing `queryClient` outside of our app — which caused the data to be shared between requests and users.

I noticed that that at one point the Next.js example initialized the `queryClient` this way and is the reason we did so (and I assume other people may have as well): https://github.com/tannerlinsley/react-query/commit/862bb2be3a7c6161b70a2310f4661859adb6d943

I think it could be really valuable to other users to alert them that their queryClient is misconfigured via a `console.warn`!

**Comment by TkDodo** [comment]
not sure how we would find that out? In a client rendered app, it is advised to create the client outside of the `App` component to have a stable reference. `useState` can also be used, but I wouldn't know under which condition we should issue the log statement...
